seedvault/src/main/java/com/solanamobile/seedvault/Bip32DerivationPath.java

```diff
@@ -19,6 +21,7 @@
  * An immutable BIP32 derivation path (see
  * <a href="https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki">BIP-0032</a>)
  */
+@RequiresApi(api = Build.VERSION_CODES.N) // library minSdk is 17
```

Comment:
Did you mean `M` for these, or specifically 24?
```suggestion
@RequiresApi(api = Build.VERSION_CODES.M) // library minSdk is 17
```

MWA has API 23, and N is 24
- https://github.com/solana-mobile/mobile-wallet-adapter/blob/main/android/walletlib/build.gradle#L17
- https://developer.android.com/reference/android/os/Build.VERSION_CODES#M

Reply:
Specifically 24. `SigningResponse` uses one method from N. Since this is really just for development and debugging purposes using the simulator, I set all classes other than `Wallet` to N for consistency.

MWA is API 23 to try and maximize compatibility, which isn't a concern here.

Reply:
That's just for the `stream()` API you're using in the `SigningResponse.toString()` right? Can that code be updated to use StringBuilder instead? It's also more performant as you won't be concatenating a new string for every signature in `mSignatures`

```java
    @NonNull
    @Override
    public String toString() {
        StringBuilder signaturesStringBuilder = new StringBuilder();
        for (int i = 0; i < mSignatures.size(); ++i) {
            signaturesStringBuilder.append(Arrays.toString(mSignatures.get(i)));
            if (i < mSignatures.size() - 1) {
                signaturesStringBuilder.append(", ");
            }
        }
        return "SigningResponse{" +
                "mSignatures=[" + signaturesStringBuilder +
                "], mResolvedDerivationPaths=" + mResolvedDerivationPaths +
                '}';
    }
```

Reply:
I think this lets us get the entire Solana Mobile Stack to be `minSDK 23` (at least for devs) which will make it a little easier for development/adoption

Reply:
Yep, can do

Reply:
Thank you!

Reply:
Done